app/namespace.go

```diff
@@ -165,277 +167,427 @@ func ReadCACerts(ctx *cli.Context) (string, error) {
-					}
-					newName := ctx.String("new-name")
-					if _, exists := n.Spec.SearchAttributes[newName]; exists {
-						return fmt.Errorf("search attribute with new name '%s' already exists", ctx.String("new-name"))
-					}
-					y, err := ConfirmPrompt(ctx, "renaming search attribute may cause failures if any worker is still using the old name of the search-attributes. confirm rename?")
-					if err != nil || !y {
-						return err
-					}
-					return c.renameSearchAttribute(ctx, n, existingName, newName)
+				{
+					Name:    "certificate-filters",
+					Usage:   "Manage optional certificate filters used by namespace to authorize client certificates based on distinguished name fields",
+					Aliases: []string{"cf"},
+					Subcommands: []*cli.Command{
+						{
+							Name:    "import",
+							Usage:   "Sets the certificate filters on the namespace",
+							Aliases: []string{"imp"},
+							Flags: []cli.Flag{
+								NamespaceFlag,
+								RequestIDFlag,
+								ResourceVersionFlag,
+								&cli.PathFlag{
+									Name:    "certificate-filter-file",
+									Usage:   `Path to a JSON file that defines the certificate filters that will be configured on the namespace. This will replace the existing filter configuration. Sample JSON: { "filters": [ { "commonName": "test1" } ] }`,
+									Aliases: []string{"file", "f"},
+								},
+								&cli.StringFlag{
+									Name:    "certificate-filter-input",
+									Usage:   `JSON that defines the certificate filters that will be configured on the namespace. This will replace the existing filter configuration. Sample JSON: { "filters": [ { "commonName": "test1" } ] }`,
+									Aliases: []string{"input", "i"},
+								},
+							},
+							Action: func(ctx *cli.Context) error {
+								fileFlagSet := ctx.Path("certificate-filter-file") != ""
+								inputFlagSet := ctx.String("certificate-filter-input") != ""
+
+								if fileFlagSet == inputFlagSet {
+									return errors.New("exactly one of the certificate-filter-file or certificate-filter-input flags must be specified")
+								}
+
+								var jsonBytes []byte
+								var err error
+
+								if fileFlagSet {
+									jsonBytes, err = ioutil.ReadFile(ctx.Path("certificate-filter-file"))
+									if err != nil {
+										return err
+									}
+								}
+
+								if inputFlagSet {
+									jsonBytes = []byte(ctx.String("certificate-filter-input"))
+								}
+
+								filters, err := parseCertificateFilters(jsonBytes)
+								if err != nil {
+									return err
+								}
+
+								n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+								if err != nil {
+									return err
+								}
+
+								fmt.Println("certificate filters before change:")
+								if err := PrintObj(fromSpec(n.Spec.CertificateFilters)); err != nil {
+									return err
+								}
+
+								n.Spec.CertificateFilters = filters.toSpec()
+
+								fmt.Println("certificate filters after change:")
```

Comment:
Do we want to maybe send a confirmation prompt before the user commits to the update? like we do for ca cert updates?

Reply:
yes, i was weighing doing this, should have asked that as an open question. I will add confirmation and an optional --force flag?